src/validators/domain.py

```diff
@@ -1,20 +1,39 @@
 """Domain."""
 
 # standard
+from os import environ
 from pathlib import Path
 import re
+from typing import Optional, Set
 
 # local
 from .utils import validator
 
 
-def _iana_tld():
-    """Load IANA TLDs as a Generator."""
-    # source: https://data.iana.org/TLD/tlds-alpha-by-domain.txt
-    with Path(__file__).parent.joinpath("_tld.txt").open() as tld_f:
-        _ = next(tld_f)  # ignore the first line
-        for line in tld_f:
-            yield line.strip()
+class _IanaTLD:
+    """Read IANA TLDs, and optionally cache them."""
+
+    _full_cache: Optional[Set[str]] = None
+    # source: https://www.statista.com/statistics/265677
+    _popular_cache = {"COM", "ORG", "RU", "DE", "NET", "BR", "UK", "JP", "FR", "IT"}
+
+    @classmethod
+    def _retrieve(cls):
+        with Path(__file__).parent.joinpath("_tld.txt").open() as tld_f:
+            _ = next(tld_f)  # ignore the first line
+            for line in tld_f:
+                yield line.strip()
+
+    @classmethod
+    def check(cls, tld: str):
+        if tld in cls._popular_cache:
+            return True
+        if not cls._full_cache:
+            if environ.get("PYVLD_CACHE_TLD", "False") == "True":
+                cls._full_cache = set(cls._retrieve())
+            else:
+                return tld in cls._retrieve()
+        return tld in cls._full_cache
```

Comment:
Simpler?

```py
    def check(cls, tld: str):
        if tld in cls._popular_cache:
            return True
        if not cls._full_cache and if environ.get("PYVLD_CACHE_TLD") != "True":
            return tld in cls._retrieve()
        cls._full_cache = set(cls._retrieve())
        return tld in cls._full_cache
```

Reply:
This always reads the file when `PYVLD_CACHE_TLD` is set, instead of just once.

Reply:
Ah my bad.